scripts/test/Muon/thread_model_test.py

```diff
@@ -0,0 +1,185 @@
+import unittest
+import sys
+
+if sys.version_info.major == 3:
+    from unittest import mock
+else:
+    import mock
+
+from PyQt4.QtGui import QApplication
+
+from Muon.GUI.Common.thread_model import ThreadModel
+
+
+class testModelWithoutExecute:
+
+    def __init__(self):
+        pass
+
+    def output(self):
+        pass
+
+
+class testModelWithoutOutput:
+
+    def __init__(self):
+        pass
+
+    def execute(self):
+        pass
+
+
+class testModelWithoutLoadData:
+
+    def __init__(self):
+        pass
+
+    def execute(self):
+        pass
+
+    def output(self):
+        pass
+
+
+class testModel:
+
+    def __init__(self):
+        self._data = None
+
+    def loadData(self, data):
+        self._data = data
+
+    def output(self):
+        pass
+
+    def execute(self):
+        pass
+
+
+class LoadFileWidgetViewTest(unittest.TestCase):
+    class Runner:
+        """This runner class creates a main event loop for threaded code to run within (otherwise the threaded
+        code will not connect signals/slots properly).
+        The finished signal of a QThread is connected to the finished method below"""
+        QT_APP = QApplication([])
```

Comment:
https://github.com/mantidproject/mantid/blob/master/scripts/Muon/GUI/Common/mock_widget.py

Reply:
I don't think this will work here. The `Runner` class is doing a very specific thing (i.e. more than just returning the Application instance). See 

https://github.com/mantidproject/mantid/pull/23371#discussion_r214669503

Reply:
And it think it is overkill to simply replace this line with a call to `mock_widget`

Reply:
it means we have everything in one place, so it is only one update

Reply:
Ok, why in `mock_widget` is it necessary to use both `QApplication.instance()` and `QApplication([])`?

Reply:
I allows a widget object to be mocked without causing the test to hang. Martyn helped me with it.

Reply:
Ah, so it prevents multiple instances of `QApplication([])` from being called within the same test suite. Alternatively, as in the load widget tests, you can create a global variable at the top of the file to hold the single call to `QApplication([])` (or my code above is simple enough to just call it once within the test class).

I have changed it to the mock widget.